### PR TITLE
feat(infra): Support warp config in infra from non-evm routes

### PR DESCRIPTION
**Pull request by Mo-Hussain**
### Description

- Add in code warp configs for non-evm chains

### Testing

Manual

### CODE REVIEW

File: typescript/infra/scripts/check-deploy.ts [hunk 1]
```diff
@@ -153,16 +153,39 @@ async function check() {
...
     const filteredAddresses = Object.keys(addresses) // filter out changes not in config
       .filter((key) => key in config)
       .reduce((obj, key) => {
         obj[key] = addresses[key];
         return obj;
       }, {} as typeof addresses);
-    const app = HypERC20App.fromAddressesMap(
-      filteredAddresses as HyperlaneAddressesMap<HypERC20Factories>,
+
+    const { contractsMap, foreignDeployments } =
```
Comment (Mo-Hussain) [inline typescript/infra/scripts/check-deploy.ts:164]: separate `foreignDeployments`

File: typescript/sdk/src/router/HyperlaneRouterChecker.ts [hunk 2]
```diff
@@ -94,13 +94,22 @@ export class HyperlaneRouterChecker<
         actualConfig = await ismReader.deriveIsmConfig(actualIsmAddress);
       }
 
+      let expectedConfig = config.interchainSecurityModule;
```
Comment (Mo-Hussain) [inline typescript/sdk/src/router/HyperlaneRouterChecker.ts:97]: Support passing in an ISM address in the in code config

File: typescript/infra/scripts/check-deploy.ts [hunk 3]
```diff
@@ -153,16 +153,39 @@ async function check() {
...
+    ) {
+      console.log(
+        `${
+          chain ?? fork
+        } is non evm and it not compatible with warp checker tooling`,
+      );
+      return;
+    }
+
+    const app = new HypERC20App(
```
Comment (Mo-Hussain) [inline typescript/infra/scripts/check-deploy.ts:184]: use `HypERC20App` constructor to pass in `foreignDeployments` as an arg

**Review by tkporter — APPROVED** [review]
Plan is to chat early next week about potential changes to these configs?

File: typescript/sdk/src/contracts/contracts.ts [hunk 4]
```diff
@@ -137,7 +137,36 @@ export function attachContractsMapAndGetForeignDeployments<
       ProtocolType.Ethereum,
       metadataManager,
     ),
-    (_, addresses) => hexOrBase58ToHex(addresses.router),
+    (chain, addresses) => {
+      const router =
+        addresses.router ||
```
[the remark this answers is not in the document]
Comment (Mo-Hussain) [reply typescript/sdk/src/contracts/contracts.ts:142]: Happy to re think this as a follow up

**Pull request merged**